src/develop/imageop.c

```diff
@@ -868,6 +868,108 @@ static void dt_iop_gui_duplicate_callback(GtkButton *button, gpointer user_data)
   dt_iop_gui_duplicate(user_data, TRUE);
 }
 
+typedef struct dt_iop_gui_rename_module_t
+{
+  GtkWidget *floating_window;
+  dt_iop_module_t *module;
+} dt_iop_gui_rename_module_t;
+
+// http://stackoverflow.com/questions/4631388/transparent-floating-gtkentry
+static gboolean _rename_module_key_press(GtkWidget *entry, GdkEventKey *event, dt_iop_gui_rename_module_t *d)
+{
+  int ended = 0;
+
+  switch(event->keyval)
+  {
+    case GDK_KEY_Escape:
+      ended = 1;
+      break;
+    case GDK_KEY_Return:
+    case GDK_KEY_KP_Enter:
+    {
+      const gchar *name = gtk_entry_get_text(GTK_ENTRY(entry));
+      if(strcmp(d->module->multi_name, name) != 0)
+      {
+        g_strlcpy(d->module->multi_name, name, sizeof(d->module->multi_name) - 1);
+        d->module->multi_name[sizeof(d->module->multi_name) - 1] = 0;
+        dt_dev_add_history_item(d->module->dev, d->module, TRUE);
+        dt_iop_gui_update_header(d->module);
+      }
+
+      ended = 1;
+    }
+    break;
+  }
+
+  if(ended)
+  {
+    gtk_widget_destroy(d->floating_window);
+    free(d);
+    return TRUE;
+  }
+  return FALSE; /* event not handled */
+}
+
+static gboolean _rename_module_destroy(GtkWidget *widget, GdkEvent *event, gpointer user_data)
+{
+  gtk_widget_destroy(GTK_WIDGET(user_data));
+  return FALSE;
+}
+
+static void _iop_gui_rename_module(dt_iop_module_t *module)
+{
+  // we don't want to rename one-instance modules
```

Comment:
Is that because it will introduce a bug? it may be interesting to rename to give a hint of what this module is used for on the history?

Reply:
Not at all, but the menu entry is in the multi-instance menu, so on one-instance modules it won't be available anyway.

Reply:
But if you like the feature I can move it to the right-click menu.
What do you like best for the entry?
rename
rename module
rename instance
(any other)